docs/birthday-kata.md

```diff
@@ -0,0 +1,890 @@
+
+### Email
+Let's apply some similar decision making to get an Email.
+```unison
+unique type Email = 
+  { to: EmailAddress
+  , subject: Subject
+  , body: Body
+  }
+
+unique type EmailAddress = EmailAddress Text
+unique type Subject = Subject Text
+unique type Body = Body Text
+```
+
+Let's also update `Employee` to have an email address
+```unison
+unique type Employee  = 
+  { id: EmployeeId
+  , name: Name
+  , birthday: Date 
+  , email: EmailAddress
+  }
+```
+
+It would be nice to add some constraints to the email address (like a regex to ensure only valid email addresses can exist) but this is pretty good so far!
+
+But maybe we can build a little bit toward the future here. The prompt told us the future will likely include sending birthday messages via other platforms, like SMS or robocalls.
+
+Maybe our model should actually be about a `Message` of which there is some relationship with an `Email`. One possible way to model this would be
+```unison
+unique type Message 
+  = Email EmailAddress Subject Body
+```
+which would give us the option of extending it in the future as needed
+```unison
+unique type Message 
+  = Email EmailAddress Subject Body
+  | SMS PhoneNumber Body
+  | TwitterDM TwitterHandle Body
+```
+
+### Summary
+There isn't one "best" way to model this, but this is one possible way:
+```unison
+unique type Employee = 
+  { id: EmployeeId
+  , name: Name
+  , birthday: Date 
+  }
+
+unique type EmployeeId = EmployeeId Text
+unique type FirstName = FirstName Text
+unique type LastName = LastName Text
+unique type Name = 
+  { first: FirstName
+  , last: LastName
+  }
+
+unique type Date = 
+  { day: Day
+  , month: Month
+  , year: Year
+  }
+unique type Month 
+  = January
+  | February
+  | March
+  | April
+  | May
+  | June
+  | July
+  | August
+  | September
+  | October
+  | November
+  | December
+unique type Day = Day Nat
+unique type Year = Year Nat
+
+unique type Message 
+  = Email EmailAddress Subject Body
+
+unique type EmailAddress = EmailAddress Text
+unique type Subject = Subject Text
+unique type Body = Body Text
+```
+
+## Abilities
+Now that we have our model, let's quickly talk about the effects/infrastructure we know we will need.
+
+One of the goals of this exercise (and generally working as a software engineer) is to decouple infrastructure from our business logic (which is often approximately the same thing as spearating pure and impure code). In OO we would use interfaces to define contracts and dependency inject in whatever implementations we need (be it postgres, aws, in memory, mocks, etc).
+
+In Unison, we leverage Abilities (also called Algebraic Effects) to accomplish something similar. See [the abilities documentation](ability-typechecking.markdown) for a more detailed overview of Abilities.
```

Comment:
I think this link could point to the hosted abilities page, which is a little more fleshed out: https://www.unisonweb.org/docs/abilities/